src/Rings/oscar_singular.jl

```diff
@@ -0,0 +1,458 @@
+  R::ZZRing # singleton, but keep for consistency
+  S::Singular.Integers
+end
+
+iso_oscar_singular_coeff_ring(R::ZZRing) = OscarSingularCoefficientRingMapZZ(R, Singular.Integers())
+
+# conversion is unambigous via parent object call
+
+# QQ
+struct OscarSingularCoefficientRingMapQQ <: OscarSingularCoefficientRingMap{QQField, Singular.Rationals}
+  R::QQField # singleton, but keep for consistency
+  S::Singular.Rationals
+end
+
+iso_oscar_singular_coeff_ring(R::QQField) = OscarSingularCoefficientRingMapQQ(R, Singular.Rationals())
+
+# prime field, small characteristic
+struct OscarSingularCoefficientRingMapfpField <: OscarSingularCoefficientRingMap{fpField, Singular.N_ZpField}
+  R::fpField
+  S::Singular.N_ZpField
+end
+
+iso_oscar_singular_coeff_ring(R::fpField) = OscarSingularCoefficientRingMapfpField(R, Singular.Fp(Int(characteristic(R))))
+
+# ZZ/nZZ, n small and big
+struct OscarSingularCoefficientRingMapZZMod{U} <: OscarSingularCoefficientRingMap{U, Singular.N_ZpField}
+  R::U
+  S::Singular.N_ZnRing
+end
+
+iso_oscar_singular_coeff_ring(R::U) where {U <: Union{zzModRing, ZZModRing}} = OscarSingularCoefficientRingMapZZMod{U}(R, Singular.residue_ring(Singular.Integers(), BigInt(modulus(R)))[1]
+)
+
+# QQ(a)
+struct OscarSingularCoefficientRingMapAbsSimpleNumField <: OscarSingularCoefficientRingMap{AbsSimpleNumField, Singular.N_ZpField}
+  R::AbsSimpleNumField
+  S::Singular.N_AlgExtField
+end
+
+function iso_oscar_singular_coeff_ring(R::AbsSimpleNumField)
+  minpoly = defining_polynomial(R)
+  Qa = parent(minpoly)
+  a = gen(Qa)
+  SQa, (Sa,) = Singular.FunctionField(Singular.QQ, _variables_for_singular(symbols(Qa)))
+  Sminpoly = SQa(coeff(minpoly, 0))
+  for i in 1:degree(minpoly)
+    Sminpoly += SQa(coeff(minpoly, i))*Sa^i
+  end
+  SK, _ = Singular.AlgebraicExtensionField(SQa, Sminpoly)
+  return OscarSingularCoefficientRingMapAbsSimpleNumField(R, SK)
+end
+
+# It is nonsense, but the conversion resides in src/number/n_algExt.jl
+# via parent object call overloading
+
+# GF(p, n), small p
+
+struct OscarSingularCoefficientRingMapfqPolyRepField <: OscarSingularCoefficientRingMap{fqPolyRepField, Singular.N_ZpField}
+  R::fqPolyRepField
+  S::Singular.N_AlgExtField
+end
+
+function iso_oscar_singular_coeff_ring(F::fqPolyRepField)
+  # TODO: the Fp(Int(char)) can throw
+  minpoly = modulus(F)
+  Fa = parent(minpoly)
+  SFa, (Sa,) = Singular.FunctionField(Singular.Fp(Int(characteristic(F))),
+                                                    _variables_for_singular(symbols(Fa)))
+  Sminpoly = SFa(coeff(minpoly, 0))
+  for i in 1:degree(minpoly)
+    Sminpoly += SFa(coeff(minpoly, i))*Sa^i
+  end
+  SF, _ = Singular.AlgebraicExtensionField(SFa, Sminpoly)
+  return OscarSingularCoefficientRingMapfqPolyRepField(F, SF)
+end
+
+# Finite field (FqField)
+#
+struct OscarSingularCoefficientRingMapFqField <: OscarSingularCoefficientRingMap{FqField, Singular.N_ZpField}
+  R::FqField
+  S#= unstable =#
+  iso::MapFromFunc{FqField, FqField}
```

Comment:
The comment indicates that the type of `S` is unstable, but above the type of the codomain is declared to be `N_ZpField`. Why not do
```suggestion
struct OscarSingularCoefficientRingMapFqField{U} <: OscarSingularCoefficientRingMap{FqField, U}
  R::FqField
  S::U
  iso::MapFromFunc{FqField, FqField}
```